connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractHerder.java

```diff
@@ -255,8 +255,8 @@ public ConfigInfos validateConnectorConfig(Map<String, String> connectorConfig)
             allGroups.addAll(basicConfigDef.groups());
 
             ConnectorConfig connectorConfigToEnrich = (connector instanceof SourceConnector)
-                    ? new SourceConnectorConfig(plugins(), connectorConfig)
-                    : new SinkConnectorConfig(plugins(), connectorConfig);
+                    ? new SourceConnectorConfig(plugins(), connectorConfig, false)
```

Comment:
I don't think it's a matter of configDef being public. (They seem to already are). It's a matter of composing two or more `ConfigDef`s together. Which is what `embed` does. But the name is misleading.